### ISSUE TITLE
Display Unicode text

### ISSUE THREAD

**Issue by hamedheidarian** [msharov/snownews#33]
Hello, I know I have to use -l flag to force snownews to use a specific charset; but can anyone tell me exatctly how i can use UTF-8?

**Comment by kouya** [comment]
The charset of your terminal should be picked up automatically. Forcing a different one will most likely mess up things. What are you trying to solve? If the display is messed up, can you add a screenshot example?

**Comment by hamedheidarian** [comment]
these characters are shown instead of farsi characters.
![screenshot from 2019-01-28 14-39-13](https://user-images.githubusercontent.com/22965390/51832987-6fdede00-230b-11e9-95ce-3e666f9fd8e9.png)

**Comment by kouya** [comment]
Oh, that's a tough one. It's also right-to-left (correct?).

I have no experience with this. But everything does work if I try cyrillic at least.

What's the locale of the system you are using set to? (command line: locale) And iconv does support that one? (aka it is contained in the output of: iconv -l) And finally the terminal itself does work if you for example cat a text file containing a text in that language?

**Comment by hamedheidarian** [comment]
it's right to left. my locale is en_US.UTF-8 and its not listed in output of iconv -l. my terminal emulator shows farsi characters correctly and also supports right to left text.

**Comment by kouya** [comment]
Can you give me the feed's URL?

**Comment by hamedheidarian** [comment]
http://twitrss.me/twitter_user_to_rss/?user=SharghDaily

**Comment by programings** [comment]
Same problem with Cyrillic text. Feed's URL: https://vasil.ludost.net/blog/?feed=rss2
Locale: LANG=en_US.UTF-8

**Comment by programings** [comment]
Same issue with Cyrillic text. Feed's URL: https://vasil.ludost.net/blog/?feed=rss2
Locale: en_US.UTF-8

**Comment by msharov** [comment]
I reviewed available resources for this and while the display of all european languages can be implemented without too much difficulty using libICU (which snownews already links to through libxml2), displaying scripts like farsi is insanely difficult and so pretty much out of the question. Existing libraries for layout, such as harfbuzz, only work on graphically displayed glyphs. libICU has a layout engine, without much documentation and only in C++. Using it will require converting snownews to C++, which I may consider doing in the future.

The bottom line for this issue is: I'll implement support for Cyrillic and the like, but all the weird scripts will be wontfix.

**Comment by msharov** [comment]
snownews will now display UTF8 text, but only for languages with one character per cell. Nonspacing characters or complex layout rules require a full layout engine. I was unable to find a suitable engine to use here and I am unqualified to write one, so for now this is the best I can do.
